lib/new_relic/agent/pipe_channel_manager.rb

```diff
@@ -85,12 +90,28 @@ def deserialize_message_length(data)
         end
 
         def write(data)
+          lock_granted = @write_lock.try_lock
```

Comment:
Why do you think it's the thing to do? In my mind we want to know if multiple thread are writing to the pipe. This code will only detect if two thread are writing concurrently,